core/api/service/api_service.cpp

```diff
@@ -51,80 +62,119 @@ namespace kagome::api {
 
     BOOST_ASSERT(app_state_manager);
     app_state_manager->takeControl(*this);
+
+    BOOST_ASSERT(subscription_engines_.events);
+    BOOST_ASSERT(subscription_engines_.storage);
   }
 
   bool ApiService::prepare() {
     for (const auto &listener : listeners_) {
-      auto on_new_session = [wp = weak_from_this()](
-                                const sptr<Session> &session) mutable {
-        auto self = wp.lock();
-        if (!self) {
-          return;
-        }
+      auto on_new_session =
+          [wp = weak_from_this()](const sptr<Session> &session) mutable {
+            auto self = wp.lock();
+            if (!self) {
+              return;
+            }
+
+            if (SessionType::kWs == session->type()) {
+              auto session_context =
+                  self->storeSessionWithId(session->id(), session);
+              session_context.storage_subscription->setCallback(
+                  [wp](uint32_t set_id,
+                       SessionPtr &session,
+                       const auto &key,
+                       const auto &data,
+                       const auto &block) {
+                    if (auto self = wp.lock()) {
+                      jsonrpc::Value::Array out_data;
+                      out_data.emplace_back(api::makeValue(key));
+                      out_data.emplace_back(api::makeValue(data));
+
+                      /// TODO(iceseer): PRE-475 make event notofication
+                      /// depending in blocks, to butch them in a single message
```

Comment:
```suggestion
                      /// TODO(iceseer): PRE-475 make event notification
                      /// depending in blocks, to batch them in a single message
```
I guess. Also, the meaning is not very clear to be honest.

Reply:
fixed